tests/test_aliases.py

```diff
@@ -121,6 +121,18 @@ def test_recursive_callable_partial_none(xonsh_execer, xonsh_builtins):
         "echo 'hi' or echo 'there'",
         "echo 'hi' && echo 'there'",
         "echo 'hi' || echo 'there'",
+        "echo --version and echo 'there'",
+        "echo --version && echo 'there'",
+        "echo --version or echo 'there'",
+        "echo --version || echo 'there'",
+        "echo -+version and echo 'there'",
+        "echo -+version && echo 'there'",
+        "echo -+version or echo 'there'",
+        "echo -+version || echo 'there'",
+        "echo -~version and echo 'there'",
+        "echo -~version && echo 'there'",
+        "echo -~version or echo 'there'",
+        "echo -~version || echo 'there'",
```

Comment:
These tests look sensible, but they **do pass even without the fix** (i.e., without the change from 04ecb2661f82612a32dd5a105b48da4aa2116e35), so it seems they test something else than what was fixed here.

Reply:
Damn it! Ho can it be? These should have triggered the NameError...
Thanks for pointing that out.

Reply:
Maybe its different within an alias than when the command line is issued explicitly? Just guessing here, I don't yet know the innards of xonsh.

Reply:
Should be OK now. Thanks for the review.